server/planning/feed_parsers/events_ml_test.py

```diff
@@ -104,5 +110,130 @@ def test_content(self):
         self.assertEqual(item["subject"], [])
 
         self.assertEqual(item["dates"]["tz"], self.app.config["DEFAULT_TIMEZONE"])
-        self.assertEqual(item["dates"]["start"], datetime(2022, 7, 5, 18, 0, tzinfo=tzoffset(None, 10800)))
-        self.assertEqual(item["dates"]["end"], datetime(2022, 7, 5, 19, 0, tzinfo=tzoffset(None, 10800)))
+        self.assertEqual(item["dates"]["start"], datetime(2022, 7, 5, 15, 0, tzinfo=utc))
+        self.assertEqual(item["dates"]["end"], datetime(2022, 7, 5, 16, tzinfo=utc))
+
+    def test_get_datetime_str_parts(self):
+        parser = EventsMLParser()
+        get_dt_str = parser.get_datetime_str
+        tz = "Europe/Prague"
+
+        self.assertEqual("2022-07-05T18:00:00+03:00", get_dt_str("2022-07-05T18:00:00+03:00", "00:00:00", tz))
+        self.assertEqual("2022-07-05T18:00:00Z", get_dt_str("2022-07-05T18:00:00Z", "00:00:00", tz))
+        self.assertEqual("2022-07-05T18:00:00+02:00", get_dt_str("2022-07-05T18:00:00", "00:00:00", tz))
+        self.assertEqual("2022-07-05T00:00:00+02:00", get_dt_str("2022-07-05", "00:00:00", tz))
+
+    def test_parse_event_schedule(self):
+        self._load_fixture("events_ml_259625.xml")
+        parser = EventsMLParser()
+        item = {}
+
+        def get_item_dates(start: str, end: Optional[str] = None):
+            root = self.xml.getroot()
+            parser.root = root
+
+            dates = root.find(parser.qname("concept")).find(parser.qname("eventDetails")).find(parser.qname("dates"))
+            for child in list(dates):
+                dates.remove(child)
+
+            etree.SubElement(dates, parser.qname("start")).text = start
+            if end is not None:
+                etree.SubElement(dates, parser.qname("end")).text = end
+
+            item.clear()
+            parser.parse_event_schedule(dates, item)
+            return item["dates"]
+
+        # Full start/end date supplied, including UTC offset
+        self.assertEqual(
+            get_item_dates("2022-07-05T18:00:00+03:00", "2022-07-05T20:00:00+03:00"),
+            dict(
+                start=datetime(2022, 7, 5, 15, 0, tzinfo=utc),
+                end=datetime(2022, 7, 5, 17, 0, tzinfo=utc),
+                tz=self.app.config["DEFAULT_TIMEZONE"],
+            ),
+        )
+
+        # Only start date & time supplied, with time NOT midnight in local time
+        self.assertEqual(
+            get_item_dates("2022-07-05T18:00:00+03:00"),
+            dict(
+                start=datetime(2022, 7, 5, 15, 0, tzinfo=utc),
+                end=datetime(2022, 7, 5, 16, 0, tzinfo=utc),
```

Comment:
I'm not sure if we should interpret this as 1h event, imo would be better to have explicit end=None, once ingested we would loose that information

Reply:
We have a config in the system for the default Event duration (the system expects start & end dates for all Events).

https://github.com/superdesk/superdesk-planning/blob/develop/server/planning/feed_parsers/events_ml.py#L133
This same config is used in the front end as well as the back end.

Reply:
yep but that doesn't really reflect the reality if it's unknown, or do we show in the UI it's using the default and we don't know the actual length?

Reply:
We're always storing an end date

Reply:
It's untested if having end=None will be supported at this time

Reply:
ok np, if that's how it works now we should use that